storage/filesystem/object.go

```diff
@@ -71,7 +71,14 @@ func (s *ObjectStorage) Get(h core.Hash) (core.Object, error) {
 	p := packfile.NewParser(r)
 
 	obj := s.NewObject()
-	return obj, p.FillObject(obj)
+	err = p.FillObject(obj)
+	if err != nil {
+		return nil, err
+	}
+	if core.AnyObject != t && obj.Type() != t {
```

Comment:
Sorry, you were right, obj is needed before checking and reading ahead for the type is convoluted.

Just leave it like it is right now, I think it is the best solution.